aws/macie.go

```diff
@@ -1,109 +1,187 @@
+	return memberAccountIds, nil
+}
+
+func removeMacieMembers(svc *macie2.Macie2, memberAccountIds []*string) error {
+
+	// Member accounts must first be disassociated
+	for _, accountId := range memberAccountIds {
+		_, err := svc.DisassociateMember(&macie2.DisassociateMemberInput{Id: accountId})
+		if err != nil {
+			return err
+		}
+		logging.Logger.Debugf("%s member account disassociated", *accountId)
 
-	return allMacieAccounts, nil
+		// Once disassociated, member accounts can be deleted
+		_, err = svc.DeleteMember(&macie2.DeleteMemberInput{Id: accountId})
+		if err != nil {
+			return err
+		}
+		logging.Logger.Debugf("%s member account deleted", *accountId)
+	}
+	return nil
 }
 
-func nukeAllMacieMemberAccounts(session *session.Session, identifiers []string) error {
+func nukeMacie(session *session.Session, identifier []string) error {
 	svc := macie2.New(session)
 	region := aws.StringValue(session.Config.Region)
 
-	if len(identifiers) == 0 {
-		logging.Logger.Debugf("No Macie member accounts to nuke in region %s", *session.Config.Region)
+	if len(identifier) == 0 {
+		logging.Logger.Debugf("No Macie member accounts to nuke in region %s", region)
 		return nil
 	}
 
-	logging.Logger.Debugf("Deleting Macie account membership and disabling Macie in %s", region)
-
-	for _, accountId := range identifiers {
-		_, disassociateErr := svc.DisassociateFromAdministratorAccount(&macie2.DisassociateFromAdministratorAccountInput{})
-
-		if disassociateErr != nil {
+	// Check for and remove any member accounts in Macie
+	// Macie cannot be disabled with active member accounts
+	memberAccountIds, err := getAllMacieMembers(svc)
+	if err != nil {
+		telemetry.TrackEvent(commonTelemetry.EventContext{
+			EventName: "Error finding macie member accounts",
+		}, map[string]interface{}{
+			"region": *svc.Config.Region,
+			"reason": "Error finding macie member accounts",
+		})
+	}
+	if err == nil && len(memberAccountIds) > 0 {
+		err = removeMacieMembers(svc, memberAccountIds)
+		if err != nil {
+			logging.Logger.Errorf("[Failed] Failed to remove members from macie")
 			telemetry.TrackEvent(commonTelemetry.EventContext{
-				EventName: "Error Nuking MACIE",
+				EventName: "Error removing members from macie",
 			}, map[string]interface{}{
-				"region": *session.Config.Region,
+				"region": *svc.Config.Region,
+				"reason": "Unable to remove members",
 			})
-			return errors.WithStackTrace(disassociateErr)
 		}
+	}
 
-		_, err := svc.DisableMacie(&macie2.DisableMacieInput{})
-
-		// Record status of this resource
-		e := report.Entry{
-			Identifier:   accountId,
-			ResourceType: "Macie member account",
-			Error:        err,
+	// Check for an administrator account
+	// Macie cannot be disabled with an active administrator account
+	adminAccount, err := svc.GetAdministratorAccount(&macie2.GetAdministratorAccountInput{})
+	if err != nil {
+		if strings.Contains(err.Error(), "there isn't a delegated Macie administrator") {
```

Comment:
I thought about using it, unfortunately it's just a generic ResourceNotFoundException (which is the same error that gets returned if Macie isn't enabled). I figured it might be better to key off of the specific message in this case, but I am definitely open to going the other way, I could see AWS potentially changing the error message breaking this.